salt/modules/zypper.py

```diff
@@ -1080,16 +1098,41 @@ def upgrade(refresh=True):
     .. code-block:: bash
 
         salt '*' pkg.upgrade
+        salt '*' pkg.upgrade dist-upgrade=True fromrepo="MyRepoName" novendorchange=True
+        salt '*' pkg.upgrade dist-upgrade=True dryrun=True
     '''
     ret = {'changes': {},
            'result': True,
            'comment': '',
     }
 
+    cmd_update = ['dist-upgrade'] if dist_upgrade else ['update']
+    cmd_update.extend(['--auto-agree-with-licenses'])
+
     if refresh:
         refresh_db()
+
+    if dryrun:
+        cmd_update.extend(['--dry-run'])
+
+    if dist_upgrade:
+        if dryrun:
+            # Creates a solver test case for debugging.
+            log.info('Executing debugsolver and performing a dry-run dist-upgrade')
+            cmd_debugsolver = cmd_update + ['--debug-solver']
+            __zypper__.noraise.call(*cmd_debugsolver)
```

Comment:
As I understand this is an extra-call in case of dry-run, additionally to that one below, right?